### PR TITLE
Discord RPC

### PR DESCRIPTION
**Pull request by chanonlim**
This is a Discord RPC implementation for Net64+  
When used and connected to a server, it should look like this:

RPC Demo Image

There may be a lot of warnings thrown though as Promises **aren't** handled.  
And also, if you want, you can send the correct artwork for gamemodes as the ones used are converted from SVG and they may be inaccurate.  
The current artworks used are:

Assets collection

Send it at SuperNiintendo#3700 and I'll upload them to Discord.  
NOTE: The artworks sent **must** be either in .png, .jpg, or .jpeg,  
**must** be at least 512x512 in size,
**highly should** be a square,
and **highly should** be a transparent image.  
The first 2nd are directly Discord's limitation, the latters are based on how Discord displays the artwork.

### CODE REVIEW

**Comment by Tarnadas** [comment]
Hey,
thanks for the PR! I'll look into it.
I haven't yet set up CI, so be aware, that I might comment on a few things

File: src/main/Connection.ts [hunk 1]
```diff
@@ -23,7 +23,51 @@ import {
 } from '../../proto/ServerClientMessage'
 
 const UPDATE_INTERVAL = 32
+const MAX_SERVER_PLAYER = 24
 
+/**
+ * Helper function to sort gamemode as the DRPC image and gamemode as readable name
+ * @param {number} gamemodeInteger - The integer of the gamemode ranging from 1-6, 8
+ */
+
```
Comment (Tarnadas) [inline src/main/Connection.ts]: remove blank line between JSDoc and function declaration

File: src/main/Connection.ts [hunk 2]
```diff
@@ -235,6 +280,16 @@ export class Connection {
         players,
         passwordRequired: handshake.passwordRequired
       })
+      var gamemodeObj = getGameModeString(handshake.gameMode)
```
Comment (Tarnadas) [inline src/main/Connection.ts]: never use var.
Either use const if possible, otherwise let

File: src/main/Connection.ts [hunk 3]
```diff
@@ -23,7 +23,51 @@ import {
...
 
 const UPDATE_INTERVAL = 32
+const MAX_SERVER_PLAYER = 24
 
+/**
+ * Helper function to sort gamemode as the DRPC image and gamemode as readable name
+ * @param {number} gamemodeInteger - The integer of the gamemode ranging from 1-6, 8
+ */
+
+function getGameModeString(gamemodeInteger: number) {
```
Comment (Tarnadas) [inline src/main/Connection.ts]: it would be much better to declare the return type. Otherwise it returns 'any'

File: src/main/index.ts [hunk 4]
```diff
@@ -48,7 +49,13 @@ export const deleteConnection = () => {
   if (connection) connection.disconnect()
   connection = undefined
 }
-
+export const RPCClient = new RPCClientInstance('550708311582834700')
+export var RPCState = {}
```
Comment (Tarnadas) [inline src/main/index.ts]: no var

**Comment by chanonlim** [comment]
> What are the unhandled Promises?
> Can you please add error handling?

Where should I log it? `console.error()` if built in dev mode?

**Comment by chanonlim** [comment]
Also I have Yarn but I normally use NPM. Should I run `yarn install` to update `yarn.lock`?

**Comment by Tarnadas** [comment]
Yes, please use Yarn for downloading dependencies. It just ensures, that you have the exact same dependencies installed.
I think that PR is good enough. There is still room to improvement, but I can look into it as well.
Also please be patient, that I won't yet make another release, but with the next one, your changes will be shipped.

**Comment by chanonlim** [comment]
OK, Thanks!

**Comment by chanonlim** [comment]
Also, the artwork has to be added in Rich Presence > Art Assets if you want to use them.
They are directly hardcoded so you can see the name and you can remove it if you'd like to.

**Pull request merged**